### ISSUE TITLE
Memory leak (of sorts) on predictions

### ISSUE THREAD

**Issue by kylebgorman** [CUNY-CL/udtube#46]
Idiotically, the prediction loop appends the predictions (the values returned by `predict_step` in the model...these are logits) to a huge list in CPU memory [here](https://github.com/Lightning-AI/pytorch-lightning/blob/4e8cf85b0cd5128adcec3f3ad0f2254f417ae1ee/src/pytorch_lightning/loops/dataloader/prediction_loop.py#L103),so if you have a long-running prediction you will eventually run out of CPU memory. 

A temporary solution is to set `--return_predictions False` (this actually doesn't break prediction, believe it or not, it just doesn't save them in the big list), but we should make it so this is automatically enabled when running in `predict` mode. This is arguably also related to #41.
